web_client.py

```diff
@@ -56,19 +53,41 @@ def _read(self, response):
         return finished
 
 
+class FakeServerProtocol(Protocol):
+
+    def __init__(self):
+        self._data = b""
+
+
+    def dataReceived(self, data):
+        self._data += data
+
+        if self._data[-4:] == b"\r\n\r\n":
+            self.transport.write(self.factory.response)
+
+        self.transport.loseConnection()
+
+
 
 interface = 0
 def main(reactor, duration):
     global interface
     concurrency = 10
 
-    root = Resource()
-    root.putChild(b'', Data(b"Hello, world", "text/plain"))
+    protoFactory = Factory.forProtocol(FakeServerProtocol)
+    protoFactory.response = b"\r\n".join(b"""HTTP/1.1 200 OK
+Date: Sat, 31 Dec 2016 07:48:22 GMT
+Connection: close
+Content-Type: text/plain
+Content-Length: 12
+Server: FakeTwistedWeb
+
+Hello, world""".split(b"\n"))
 
     interface += 1
     interface %= 255
     port = reactor.listenTCP(
-        0, Site(root), backlog=128, interface='127.0.0.%d' % (interface,))
+        0, protoFactory, backlog=128, interface='127.0.0.%d' % (interface,))
```

Comment:
A more complete separation would be achieved by running the web server in a separate process.